src/contexts/alert_handler.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useContext, useState } from "react";
+
+const AlertHandlerContext = createContext();
+
+export const AlertHandlerProvider = ({ children }) => {
+  const [alertState, setAlertState] = useState({
+    visible: false,
+    errorText: "",
+  });
+
+  const setAlert = (errorText) => {
+    const alertStateCpy = Object.assign({}, alertState);
+    alertStateCpy.visible = true;
+    alertStateCpy.errorText = errorText;
+    setAlertState(alertStateCpy);
+  };
```

Comment:
スプレッド構文を使用してみるのもいいかもしれませんね。
```suggestion
const setAlert = errorText => {
    setAlertState(prevState => ({
      ...prevState,
      visible: true,
      errorText: errorText,
    }));
  };
```